### PR TITLE
Fixes Contains.Item().And.Contains()

### LINKED ISSUE #2386: Contains.Item() fails for collections in NUnit 3.8
Hi,

just updated from NUnit 3.7.1 to 3.8.0.
Now, a test starts to fail consistently which succeeded in the past.

Repro:

```

        [Test]
        public void Collection_regression_in_380() => Assert.That(new[] { "test", "test2", "test3" }, Contains.Item("test").And.Contains("test2").And.Contains("test3"));
```

Best regards,
Ralf Koban

### PR DESCRIPTION
**Pull request by rprouse**
Fixes #2386 

This PR is being merged into release in preparation for a 3.8.1 hotfix release. I will create a separate PR to merge all changes from the hotfix back into master.

### CODE REVIEW

File: src/NUnitFramework/framework/Contains.cs [hunk 1]
```diff
@@ -37,10 +37,8 @@ public class Contains
         /// Returns a new <see cref="EqualConstraint"/> checking for the
```
Comment (jnm2) [inline src/NUnitFramework/framework/Contains.cs]: XML doc needs to be updated

File: src/NUnitFramework/framework/Does.cs [hunk 2]
```diff
@@ -63,20 +63,7 @@ public static FileOrDirectoryExistsConstraint Exist
         /// Returns a new <see cref="EqualConstraint"/> checking for the
```
Comment (jnm2) [inline src/NUnitFramework/framework/Does.cs]: XML doc also out of date

File: src/NUnitFramework/framework/Does.cs [hunk 3]
```diff
@@ -63,20 +63,7 @@ public static FileOrDirectoryExistsConstraint Exist
...
-        /// <summary>
-        /// Returns a new ContainsConstraint. This constraint
-        /// will, in turn, make use of the appropriate second-level
-        /// constraint, depending on the type of the actual argument. 
-        /// This overload is only used if the item sought is a string,
-        /// since any other type implies that we are looking for a 
-        /// collection member.
-        /// </summary>
-        public static ContainsConstraint Contain(string expected)
+        public static ContainsConstraint Contain(object expected)
```
Comment (jnm2) [inline src/NUnitFramework/framework/Does.cs]: I think we should keep the string overload for the sake of the XML docs, don't you?

File: src/NUnitFramework/tests/Constraints/CollectionEqualsTests.cs [hunk 4]
```diff
@@ -72,5 +75,32 @@ public void HonorsIgnoreCase( IEnumerable expected, IEnumerable actual )
...
+            var collection = new[] { 1, 2, 3 };
+            Assert.That(collection, Does.Contain(1).And.Contains(2).And.Contains(3));
+        }
+
+        [Test]
+        public void ContainsMultipleItemsInt()
+        {
+            var collection = new[] { 1, 2, 3 };
+            Assert.That(collection, Contains.Item(1).And.Contains(2).And.Contains(3));
+        }
```
Comment (jnm2) [inline src/NUnitFramework/tests/Constraints/CollectionEqualsTests.cs]: Tests look good, but seems weird to put them in `CollectionEqualsTests`. Do we have other tests on (collection) Contains already?

File: src/NUnitFramework/framework/Does.cs [hunk 5]
```diff
@@ -63,23 +63,8 @@ public static FileOrDirectoryExistsConstraint Exist
...
-        /// will, in turn, make use of the appropriate second-level
-        /// constraint, depending on the type of the actual argument. 
-        /// This overload is only used if the item sought is a string,
-        /// since any other type implies that we are looking for a 
-        /// collection member.
-        /// </summary>
-        public static ContainsConstraint Contain(string expected)
-        {
-            return new ContainsConstraint(expected);
-        }
```
Comment (jnm2) [inline src/NUnitFramework/framework/Does.cs]: I think we should keep the string overload for the sake of intellisense.

File: src/NUnitFramework/framework/Contains.cs (hunk 1 above)
Comment (rprouse) [reply src/NUnitFramework/framework/Contains.cs]: Good catch, thanks. I shouldn't rush 😄

File: src/NUnitFramework/tests/Constraints/CollectionEqualsTests.cs (hunk 4 above)
Comment (rprouse) [reply src/NUnitFramework/tests/Constraints/CollectionEqualsTests.cs]: Agreed, I'd looked for a better place, but didn't find one. I went back and the best I can find is a fixture called `ListContentTests` so I am moving there.

**Comment by rprouse** [comment]
@jnm2 I have pushed changes based on your review. When you get a chance, can you re-review? Thanks 👍

**Pull request merged**